omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java

```diff
@@ -0,0 +1,185 @@
+		}
+		return null;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("patient");
+		description.addRequiredProperty("encounter");
+		description.addProperty("orders");
+		description.addProperty("orderSet");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("orders");
+		return description;
+	}
+	
+	public Model getGETModel(Representation rep) {
+		ModelImpl modelImpl = (ModelImpl) super.getGETModel(rep);
+		if (rep instanceof DefaultRepresentation) {
+			modelImpl.property("uuid", new StringProperty()).property("display", new StringProperty())
+			        .property("voided", new BooleanProperty())
+			        .property("patient", new RefProperty("#/definitions/PatientGetRef"))
+			        .property("encounter", new RefProperty("#/definitions/EncounterGetRef"))
+			        .property("orders", new RefProperty("#/definitions/OrderGetRef"))
+			        .property("orderSet", new RefProperty("#/definitions/OrdersetGetRef"));
+			
+		} else if (rep instanceof FullRepresentation) {
+			modelImpl.property("uuid", new StringProperty()).property("display", new StringProperty())
+			        .property("voided", new BooleanProperty()).property("auditInfo", new BooleanProperty())
+			        .property("patient", new RefProperty("#/definitions/PatientGetRef"))
+			        .property("encounter", new RefProperty("#/definitions/EncounterGetRef"))
+			        .property("orders", new ArrayProperty(new RefProperty("#/definitions/OrderGetRef")))
+			        .property("orderSet", new RefProperty("#/definitions/OrdersetGetRef"));
+		} else if (rep instanceof RefRepresentation) {
+			modelImpl.property("display", new StringProperty()).property("uuid", new StringProperty());
+			
+		}
+		return modelImpl;
+	}
+	
+	@Override
+	public Model getCREATEModel(Representation representation) {
+		ModelImpl model = new ModelImpl().property("patient", new StringProperty().example("uuid"))
+		        .property("encounter", new StringProperty().example("uuid"))
+		        .property("orders", new ArrayProperty(new RefProperty("#/definitions/OrderCreate")))
+		        .property("orderSet", new StringProperty().example("uuid"));
+		return model;
+	}
+	
+	@Override
+	public Model getUPDATEModel(Representation rep) {
+		ModelImpl model = new ModelImpl().property("orders",
+		    new ArrayProperty(new RefProperty("#/definitions/OrderCreate")));
+		return model;
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(OrderGroup group) {
+		if (group.isVoided())
+			return "[No Group]";
+		Patient patient = group.getPatient();
+		return patient.getPatientIdentifier().getIdentifier() + " - " + patient.getPersonName().getFullName() + " - "
```

Comment:
Did you get this from a specification (how to build the display name) some where?

Reply:
yes, i tried to follow the related talk thread ,and it seems some thing like that was appropriate for the display name. though i asked for clarification on it and i got no reply